doc/requirements/sdid.rst

```diff
@@ -0,0 +1,261 @@
+
+Reactor Overview
+----------------
+
+..
+   TODO
+
+
+------
+Design
+------
+
+TODO
+
+Blueprints Design
+-----------------
+
+The ``blueprints`` package makes extensive use of the ``yamlize`` library, which provides a mechanism for serializing and deserializing blueprints to and from YAML files.
+
+
+Settings Design
+---------------
+
+Dictionary-like behavior
+^^^^^^^^^^^^^^^^^^^^^^^^
+The settings system functions much like a Python dictionary for most of its interactions.
+
+The two objects to note in this interplay are the ``Settings`` object which is what the vast majority of interactions are directed at. Then there's the more hidden ``Setting`` object which corresponds to a single defined entity in the settings system, housed under a ``Settings.settings`` dictionary.
+
+When users perform lookups and value assignments to a key on the settings object, it hides a bit of the work behind the scenes of applying setting rules.
+
+The lookup fashioned as::
+
+	>>>Settings['mySettingName']
+	1
+
+will only return the current value of the setting stored under ``mySettingName``, whereas::
+
+	>>>Settings.settings['mySettingName']
+	<IntSetting mySettingName value:1 default:1>
+
+returns the setting object itself. Only more complex relationships with coupled code tools will be concerned with
+how the object itself behaves. A prime example of this is the GUI.
+
+Improve safety around possible misuses and accidents of the settings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+While simplicity remains a principle concern, some allotment for error prevention has to be allowed. A basic system that can be misused wildly is inherently more complex than one with a little overhead that works to keep nasty surprises as bay.
+
+The settings system is written in a defensive fashion, as it's the most frequent location for possible developer and user misuse. Given this, any method from accessing non-existent settings to trying to supply duplicate settings has been written with fail-fast behavior.
+
+In some cases there may be an automated solution to a possible misuse. In general there's nothing overtly wrong
+with the litany of checks for safety but it does make the system appear more convoluted to newcomers. Of course,
+This small cost is outweighed by the time saved from working through the same resolvable problem in use many times.
+
+There will always be a need for knowledgable users to communicate with however, as it is unreasonable to attempt to
+program in a fool-proof understanding of the physics coded for in the settings system. Such an endeavor would come with great developmental and maintenance costs.
+
+Most problems these days arise when users make typos or simple editing mistakes when manually adjusting some values in the case settings input files.
```

Comment:
Again, this is a subjective claim that does not appear, to me, to belong in this document.

Reply:
Removed a few of those sentences.